OADP-4001 Self-Service (#89043)
openshift/openshift-docs: modules/oadp-self-service-unsupported-features.adoc

// Module included in the following assemblies:
//
// backup_and_restore/application_backup_and_restore/oadp-self-service/oadp-self-service.adoc

:_mod-docs-content-type: CONCEPT
[id="oadp-self-service-unsupported-features_{context}"]
= {oadp-short} Self-Service unsupported features

The following features are not supported by {oadp-short} Self-Service:

* Cross cluster backup and restore, or migrations are not supported. These {oadp-short} operations are supported for the cluster administrator.

* A non-admin user cannot create a `VolumeSnapshotLocation` (VSL) CR. The cluster administrator creates and configures the VSL in the `DataProtectionApplication` (DPA) for a non-admin user.

* `ResourceModifiers` and volume policies are not supported for a non-admin user.

> * A non-admin user cannot request backup and restore logs by using the `NonAdminDownloadRequest` CR for a `BackupStorageLocation` created by the cluster administrator. A non-admin user must create a `NonAdminBackupStorageLocation` in their authorized namespace to access the backup and restore logs.

Reviewer: This sentence is too complex, maybe reverse it's logic:

A non-admin user can request backup or restore logs using the NonAdminDownloadRequest CR, but only if the backups or restores were created using NonAdminBackupStorageLocations defined by the user and not the cluster-wide default BSL.

Author: re-worded. thanks.

* To ensure secure backup and restore, Self-Service automatically excludes the following resources from being backed up or restored:

> ** Security Context Constraints (SCCs)

Author: done.

** Cluster roles
** Cluster role bindings

> ** Custom resource definitions (CRDs)

Author: done.

// Module included in the following assemblies:
//
// backup_and_restore/application_backup_and_restore/oadp-self-service/oadp-self-service.adoc

:_mod-docs-content-type: CONCEPT
[id="oadp-self-service-unsupported-features_{context}"]

> = {oadp-short} Self-Service unsupported features

Reviewer 2: Just musing here, but to me "unsupported features" implies that you _can_ do these things but they are unsupported. Or to put it another way, these are something like upstream features that RH does not support downstream. Perhaps calling these unsupported anti-patterns "limitations" might be clearer?

I could be wrong, so please feel free to ignore this comment if it isn't useful. It is only something to consider.

Author: renamed to `limitations`

The following features are not supported by {oadp-short} Self-Service:

* Cross cluster backup and restore, or migrations are not supported. These {oadp-short} operations are supported for the cluster administrator.

> * A namespace admin user cannot create a `VolumeSnapshotLocation` (VSL) CR. The cluster administrator creates and configures the VSL in the `DataProtectionApplication` (DPA) CR for a namespace admin user.

Reviewer 3: ```suggestion
* A namespace `admin` user cannot create a `VolumeSnapshotLocation` (VSL) CR. The cluster administrator creates and configures the VSL in the `DataProtectionApplication` (DPA) CR for a namespace `admin` user.
```

- Expand CR here for its first instance

> * `ResourceModifiers` CR and volume policies are not supported for a namespace admin user.

Reviewer 3: ```suggestion
* The `ResourceModifiers` CR and volume policies are not supported for a namespace `admin` user.
```

> * A namespace admin user can request backup or restore logs by using the `NonAdminDownloadRequest` CR, only if the backup or restore is created by using the `NonAdminBackupStorageLocation` CR created by the user and not the cluster-wide default backup storage location.

Reviewer 3: ```suggestion
* A namespace `admin` user can request backup or restore logs by using the `NonAdminDownloadRequest` CR, only if the backup or restore is created by a user through the `NonAdminBackupStorageLocation` CR and not the cluster-wide default backup storage location.
```

> * A namespace `admin` user can request backup or restore logs by using the `NonAdminDownloadRequest` CR, only if the backup or restore is created by a user through the `NonAdminBackupStorageLocation` CR and not the cluster-wide default backup storage location.

Reviewer 2: This sentence is pretty confusing and very long. If you get the chance, consider breaking it in two and clarifying it. I am not sure if this changes the meaning, but I took a stab at it:

```suggestion
* If the backup or restore operation is created through the cluster-wide default backup storage location, a namespace `admin` user cannot request backup or restore logs. If you want a namespace `admin` user to be able to request backup or restore logs by using a `NonAdminDownloardRequest` CR, the backup or restore operation must be created by using the `NonAdminBackupStorageLocation` CR. 
```
Not a blocker.

Author: refactored the sentence

> * To ensure secure backup and restore, {oadp-short} Self-Service automatically excludes the following custom resources from being backed up or restored:

Reviewer 3: ```suggestion
* To ensure secure backup and restore, {oadp-short} Self-Service automatically excludes the following CRs from being backed up or restored:
```
